src/service/authentication/OrganisationAuthenticationRepository.ts

```diff
@@ -0,0 +1,20 @@
+import { PasswordIndex } from "./BasicAuthenticationMiddleware";
+import { keyValue } from "ts-array-utils";
+
+export class OrganisationAuthenticationRepository {
+
+  constructor(
+    private readonly db: any
+  ) {}
+
+  public async getPasswordIndex(): Promise<PasswordIndex> {
+    const [rows]: CredentialRow[][] = await this.db.query("SELECT id, api_key FROM organisation");
+
+    return rows.reduce(keyValue(row => [row.id, Buffer.from(row.api_key).toString()]), {});
+  }
+}
+
+interface CredentialRow {
+  id: number,
+  api_key: Buffer
```

Comment:
why is it a buffer here but a string in the organisation interface object

Reply:
The column type is binary so when the mysql2 library loads it out of the DB it uses a Buffer. I believe it's to avoid character encoding issues.